### PR TITLE
Make MQTT the default transport

### PR DESCRIPTION
**Pull request by drwill-ms**
Lorenzo supports making MQTT the default, and suggested we move ahead with that decision.

### CODE REVIEW

File: iothub/device/tests/Edge/EdgeModuleClientFactoryTest.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public async Task TestCreate_FromConnectionStringEnvironment_ShouldCreateClient(
         public async Task TestCreate_FromConnectionStringEnvironment_ShouldCreateClientWithOptions()
         {
             // setup
-            var clientOptions = new IotHubClientOptions
+            var clientOptions = new IotHubClientOptions(new IotHubClientAmqpSettings())
```
Comment (drwill-ms) [inline iothub/device/tests/Edge/EdgeModuleClientFactoryTest.cs:48]: Chose AMQP given this is an Edge module.

File: iothub/device/tests/Transport/Amqp/MockableAmqpUnit.cs [hunk 2]
```diff
@@ -13,17 +14,19 @@ namespace Microsoft.Azure.Devices.Client.Test.Transport
 {
     internal class MockableAmqpUnit : AmqpUnit
     {
+        private static IotHubClientAmqpSettings s_transportSettings = new();
```
Comment (drwill-ms) [inline iothub/device/tests/Transport/Amqp/MockableAmqpUnit.cs:17]: Need to explicitly initialize AMQP for this mock, and I changed the init below to all refer to the same instance.

File: iothub/device/tests/Transport/Mqtt/MqttIotHubAdapterTest.cs [hunk 3]
```diff
@@ -111,7 +111,7 @@ public void TestAuthenticationChain()
             var passwordProvider = new Mock<IAuthorizationProvider>();
             var mqttIotHubEventHandler = new Mock<IMqttIotHubEventHandler>();
             var productInfo = new ProductInfo();
-            var options = new IotHubClientOptions();
+            var options = new IotHubClientOptions(new IotHubClientMqttSettings());
```
Comment (drwill-ms) [inline iothub/device/tests/Transport/Mqtt/MqttIotHubAdapterTest.cs:114]: Oops, these tests are supposed to be for MQTT, but without being explicit they were testing AMQP before. Fixing that, even though the default is now MQTT.

**Review by timtay-microsoft — APPROVED** [review]

**Review by andyk-ms — APPROVED** [review]

**Pull request merged**